### DISCUSSION TITLE
How to trim excess linefeeds from my measures

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#588] [🙋♀️TE3 Q&A] [answered]
I have the following code to format my DAX code and write it into the "Description" field.
<code>/*  Cycle over all measures in model format using
    DAX Formatter with Short Lines, then add 1 line
    feed to the start of the measure 
    Then copies the DAX to the description so it
    is available in the hover over */

FormatDax(Model.AllMeasures, true);
foreach (var m in Model.AllMeasures)
    {
        m.Expression = '\n' + m.Expression;
        m.Description = string.Join("\n", m.Expression.Split('\n').Take(16).ToArray());
    }

/*  Below just copies DAX expression to the description of Calc Columns
    which helps when browsing fields. It will overwrite any previous
    description, so any TODO expressions need to be in the DAX comments */
foreach (var c in Model.AllColumns.OfType<CalculatedColumn>())
    {
        c.Description = string.Join("\n", c.Expression.Split('\n').Take(20).ToArray());
    }</code>

This worked fine until 3.30. Now every time I run it I am getting a new blank line inserted at the top, so after running it a few times, I have 8 or 9 blank lines at the top. 

I think my macro is working against a probable improvement in 3.3 where measures formatted in Desktop will now show the formula starting on a new line, not on the line with the measure name.

I tried using string.Trim() but that just gives me an error about strings converting to char.

So I guess I am asking for help in 2 ways:

1. a way to remove all leading blank lines that are now in my measures
2. A way to only include the first 16 lines of the measure in the description box without the string.Join("\n".... part. I tried just using the m.Expression.Split('\n').Take(16).ToArray()) code, but that also fails with cannot convert string to char, so that must also be part of the string.Join function.
Thanks.

**Comment by albertospelta** [comment, marked as answer]
Hi @edhans,
you can use `TrimStart` to remove all leading blank lines. In the following example I've also included `.Replace("\r\n", "\n")` just to be sure that the line ending characters are normalised in the formatted string.
``` c#
foreach (var m in Model.AllMeasures) 
{ 
    m.Expression = '\n' + m.Expression.Replace("\r\n", "\n").TrimStart('\n'); 
}
```
I am not sure I understand your second question. This code seems to work correctly without any conversion errors

``` c#
foreach (var m in Model.AllMeasures) 
{ 
    m.Description = string.Join("\n", m.Expression.Split('\n').Take(16));
}
```

**Reply by edhans** [reply]
Thanks @albertospelta - I think your answer in the first part - which works - also fixed my issue in the second part where additional spaces were creeping in. Much appreciated!
